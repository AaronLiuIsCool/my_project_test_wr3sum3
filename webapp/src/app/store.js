import { configureStore } from '@reduxjs/toolkit';
import resumeReducer from 'features/SmartResume/slicers/resumeSlicer';
import counterReducer from 'features/CounterExample/counterSlice.js';

export default configureStore({
  reducer: {
    counter: counterReducer,
    resume: resumeReducer
  },
});
