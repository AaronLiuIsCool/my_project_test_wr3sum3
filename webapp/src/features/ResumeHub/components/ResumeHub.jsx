import React from 'react';
import { useSelector } from 'react-redux';

import CallToAction from './CallToAction';

import { selectLanguage, selectPath } from 'features/App/slicer';
// import { getLogger } from 'shell/logger';
import { I8nContext } from 'shell/i18n';
// import ResumeServices from 'shell/services/ResumeServices';

import zh from '../i18n/zh.json';
import en from '../i18n/en.json';

// const logger = getLogger('ResumeHub');
// const resumeServices = new ResumeServices();

const renderHub = (path = {}) => {
    switch (path.search) {
        case '?hub':
            return (
                <div>show a list of resumes</div>
            );
        default:
            return <CallToAction />;
    };
}

const ResumeHub = () => {
    // const dispatch = useDispatch();
    // const [ready, setReady] = useState(false);
    const path = useSelector(selectPath);
    const language = useSelector(selectLanguage);
    const messages = language === 'zh' ? zh : en;

    return (
        <I8nContext.Provider value={messages}>
            <div className='features resumeHub'>
                {renderHub(path)}
            </div>
        </I8nContext.Provider>
    );
};

export default ResumeHub;