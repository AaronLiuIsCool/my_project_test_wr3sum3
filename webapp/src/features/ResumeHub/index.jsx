import React, { Suspense } from 'react';

import LoadFallbackComponent from 'components/LoadFallbackComponent';
import './styles/index.scss';

const ResumeHub = React.lazy(() => import('./components/ResumeHub'));

const LazyLoadResumeHub = () => (
  <Suspense fallback={<LoadFallbackComponent />}>
    <ResumeHub />
  </Suspense>
);

export default LazyLoadResumeHub;
