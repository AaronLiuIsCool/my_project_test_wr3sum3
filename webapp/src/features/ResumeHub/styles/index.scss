.resumeHub {
  a:hover {
    text-decoration: none;
  }  
}
