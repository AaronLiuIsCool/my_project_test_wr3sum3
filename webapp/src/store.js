import { configureStore } from '@reduxjs/toolkit';

import appReducer from 'features/App/slicer';
import accountReducer from 'features/App/slicer/account';
import resumeReducer from 'features/SmartResume/slicer';
import jobsMatcherReducer from 'features/JobsMatcher/slicer';

export default configureStore({
  reducer: {
    app: appReducer,
    account: accountReducer,
    resume: resumeReducer,
    jobs: jobsMatcherReducer,
  },
});
