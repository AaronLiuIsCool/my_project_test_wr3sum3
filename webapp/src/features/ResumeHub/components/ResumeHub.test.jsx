import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import store from 'store';

import ResumeHub from './ResumeHub';

describe('ResumeHub', () => {
    test('renders ResumeHub', () => {
        const resumeHub = renderer.create(
            <Provider store={store}>
                <ResumeHub />
            </Provider>
        ).toJSON();
    
        expect(resumeHub).toMatchSnapshot();
    });    
});


