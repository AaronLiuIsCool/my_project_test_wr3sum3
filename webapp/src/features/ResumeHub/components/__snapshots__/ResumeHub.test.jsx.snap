// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ResumeHub 1`] = `
<div
  className="features resumeHub"
>
  <div
    className="container"
  >
    <div
      className="illustration"
    >
      <svg>
        create_your_resume.svg
      </svg>
    </div>
    <div
      className="callToAction"
    >
      <p
        className="description"
      >
        看似您还没有简历，花5分钟创造一份吧
      </p>
      <a
        className="action btn btn-primary"
        href="/resume/new"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        开始创建
      </a>
    </div>
  </div>
</div>
`;
